Composer/packages/lib/shared/src/copyUtils/copyInputDialog.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { InputDialog } from '../types';
+
+import { ExternalApi } from './ExternalApi';
+import { shallowCopyAdaptiveAction } from './shallowCopyAdaptiveAction';
+
+export const copyInputDialog = async (input: InputDialog, externalApi: ExternalApi): Promise<InputDialog> => {
+  const copy = shallowCopyAdaptiveAction(input, externalApi);
+  const nodeId = copy.$designer ? copy.$designer.id : '';
+
+  if (input.prompt) {
```

Comment:
input.prompt looks like a requried field. so perhaps we can save some checking there?

Reply:
it's declared as required in sdk but might be empty in real dialog json. The checking here is to avoid the case that we generate a json like
```
{
  ...
  prompt: null | undefined
}
```

Reply:
So the check should not be `if (input.prompt)`, it will miss copying an empty string. Maybe `if (input.prompt !== undefined)`?